server/pulp/server/db/model/__init__.py

```diff
@@ -847,8 +859,31 @@ class LazyCatalogEntry(AutoRetryDocument):
     url = StringField(required=True)
     checksum = StringField()
     checksum_algorithm = StringField(regex=ALG_REGEX)
+    revision = IntField(default=0)
     data = DictField()
 
+    def save_revision(self):
+        """
+        Add the entry using the next revision number.
+        Previous revisions are deleted.
+        """
+        revisions = {0}
+        query = dict(
+            unit_id=self.unit_id,
+            unit_type_id=self.unit_type_id,
+            importer_id=self.importer_id)
+        # Find revisions
+        qs = LazyCatalogEntry.objects.filter(**query)
+        for revision in qs.distinct('revision'):
+            revisions.add(revision)
+        # Add new revision
+        last_revision = sorted(revisions, reverse=True)[0]
```

Comment:
Not that it matters, but you could do ``sorted(revisions)[-1]``. I'm not sure which I find more readable, though.